### PR TITLE
fix: stop blocking swap on balance fetching

### PR DESCRIPTION
**Pull request by just-toby**
https://uniswaplabs.atlassian.net/browse/WEB-2766?atlOrigin=eyJpIjoiNTA4OGVmZWI5ZTJhNDMyMThhMjkwOGRkZjc0NjBhMmUiLCJwIjoiaiJ9

we want to stop blocking swap if the balances aren't loaded to match the current behavior in interface.

if the balance IS loaded, we should still disable the button if the user doesn't have enough funds

### CODE REVIEW

File: src/components/Swap/SwapActionButton/index.tsx [hunk 1]
```diff
@@ -28,8 +28,8 @@ export default function SwapActionButton() {
       (!permit2Enabled && approval.state !== SwapApprovalState.APPROVED) ||
       error !== undefined ||
       (!isWrap && !trade) ||
-      !(inputCurrencyAmount && inputCurrencyBalance) ||
-      inputCurrencyBalance.lessThan(inputCurrencyAmount),
+      !inputCurrencyAmount ||
```
Comment (just-toby) [inline src/components/Swap/SwapActionButton/index.tsx:31]: i'm not sure if we still need this check - if we're requiring `inputCurrencyAmount`, shouldn't we also be requiring `outputCurrencyAmount`?

Comment (zzmp) [reply src/components/Swap/SwapActionButton/index.tsx:31]: We're not checking it against a balance, so we don't need to check it at all: `!(inputCurrencyAmount && inputCurrencyBalance)` acted as a guard to ensure that `inputCurrencyBalance.lessThan(inputCurrencyAmount)` was valid. I _think_ this should be covered by `(!isWrap && !trade)`.

File: src/components/Swap/SwapActionButton/index.tsx [hunk 2]
```diff
@@ -28,8 +28,8 @@ export default function SwapActionButton() {
...
+      (Boolean(inputCurrencyBalance) && Boolean(inputCurrencyBalance?.lessThan(inputCurrencyAmount))),
```
Comment (zzmp) [inline src/components/Swap/SwapActionButton/index.tsx:32]: You either need the guard check (`inputCurrencyBalance &&`) _or_ the optional operator (`?.`) - I think you can simplify it like this:

```suggestion
      inputCurrencyBalance?.lessThan(inputCurrencyAmount),
```

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/components/Swap/SwapActionButton/index.tsx (hunk 1 above)
Comment (just-toby) [reply src/components/Swap/SwapActionButton/index.tsx:31]: this is true, but we avoid some type warnings if we keep it.

File: src/components/Swap/SwapActionButton/index.tsx (hunk 2 above)
Comment (just-toby) [reply src/components/Swap/SwapActionButton/index.tsx]: still need to cast to Boolean so that `isDisabled` isn't undefined

**Review by zzmp — APPROVED** [review]

**Pull request merged**